### ISSUE TITLE
v2: reverse-proxy panic when sent http request

### ISSUE THREAD

**Issue by aca** [caddyserver/caddy#2849]
## 1. Which version of Caddy are you using (`caddy -version`)?
caddy v2 dccba71276796999f3e16a6560016dc5dd0a952a

## 2. What are you trying to do?
trying caddy reverse-proxy command

## 3. What is your Caddyfile?
none

## 4. How did you run Caddy (give the full command and describe the execution environment)?
```
 » sudo caddy reverse-proxy --from example.iptime.org --to http://localhost:8080                                        
```

## 5. Please paste any relevant HTTP request(s) here.
```
curl example.iptime.org
```

## 6. What did you expect to see?
When sending http request to the reverse proxy I see panic messages.
I think it should redirect to https properly.
Before I look into details I just want to check if is "intended" and if not, what should be done.

## 7. What did you see instead (give full error messages and/or log)?
```
1.5725370715240774e+09  INFO    http    enabling automatic TLS certificate management   {"domains": ["xnzm.iptime.org"]}
1.572537071524252e+09   INFO    http    enabling automatic HTTP->HTTPS redirects        {"domains": ["xnzm.iptime.org"]}
2019/11/01 00:51:11 [INFO][cache:0xc00013a870] Started certificate maintenance routine
2019/11/01 00:51:11 Caddy 2 proxying from http://xnzm.iptime.org to http://localhost:8080
2019/11/01 00:51:14 http: panic serving 192.168.0.1:49782: runtime error: invalid memory address or nil pointer dereference
goroutine 51 [running]:
net/http.(*conn).serve.func1(0xc0002bc140)
        /home/rok/.asdf/installs/golang/1.13.2/go/src/net/http/server.go:1767 +0x139
panic(0xccb5c0, 0x152f0c0)
        /home/rok/.asdf/installs/golang/1.13.2/go/src/runtime/panic.go:679 +0x1b2
go.uber.org/zap.(*Logger).clone(...)
        /home/rok/pkg/mod/go.uber.org/zap@v1.10.0/logger.go:252
go.uber.org/zap.(*Logger).With(0x0, 0xc000126740, 0x1, 0x1, 0xc0002c01c0)
        /home/rok/pkg/mod/go.uber.org/zap@v1.10.0/logger.go:163 +0x45
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc00021d450, 0xf1c220, 0xc0002c01c0, 0xc0002f2300)
        /home/rok/src/github.com/caddyserver/caddy/modules/caddyhttp/server.go:86 +0x677
net/http.serverHandler.ServeHTTP(0xc00019c2a0, 0xf1c220, 0xc0002c01c0, 0xc0002f2000)
        /home/rok/.asdf/installs/golang/1.13.2/go/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc0002bc140, 0xf1eea0, 0xc000126600)
        /home/rok/.asdf/installs/golang/1.13.2/go/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
        /home/rok/.asdf/installs/golang/1.13.2/go/src/net/http/server.go:2927 +0x38e

```

## Bonus: What do you use Caddy for? Why did you choose Caddy?
Personal usage, fun, study. Maybe at work if there are production level k8s ingress controller.

**Comment by aca** [comment]
```
» go version
go version go1.13.2 linux/amd64
```
@mholt , hmm.. so it is not intended. I will check in other platform/version later.

**Comment by mholt** [comment]
Right, this definitely shouldn't be panicking.

It's weird, I can't figure out why it is panicking on that line, and I cannot reproduce it even at the same commit or with the same commands as you.

The panic seems to suggest that the `srv.errorLogger` is nil, but that's almost certainly not the case: https://github.com/caddyserver/caddy/blob/8d3c64932eddb42f7e5c4549555e9d12a559d552/modules/caddyhttp/caddyhttp.go#L77-L80

Any more details to reproduce this would be appreciated, I'll keep trying though...

**Comment by mholt** [comment]
Aha! I was finally able to reproduce it. Hang tight for a fix.

**Comment by aca** [comment]
@mholt It's 3am here. I'm sorry but I'll look into it later. 
Hope you not waste your time as it might be problem with my environment...

**Comment by aca** [comment]
@mholt oh great;

**Comment by mholt** [comment]
Ha, get some sleep. I think I need some of that, too. 😅 

Fixed in fb06c04.

Again, many thanks for the reports! Keep it up! :+1:
